sdk/core/core/inc/az_credentials.h

```diff
@@ -45,12 +46,21 @@ typedef struct
 {
   struct
   {
-    uint8_t token[_az_TOKEN_BUF_SIZE]; /*!< Base64-encoded token */
     int16_t token_length;
     int64_t expires_at_msec;
+    uint8_t token[_az_TOKEN_BUF_SIZE]; /*!< Base64-encoded token */
```

Comment:
> Rearranged fields in credential structs for better alignment.

Can you please explain how this leads to better alignment?

Reply:
Ah, I should move int64_t before int16, my eyes didn't catch that initially.
Basically, on some platforms, compilers would align data to a word binary. It is usually int32, but can be any reasonable number (1, 2, 4, 8). If you put an array of 9 int8's first, and then an int32, it goes as `[int8][int8][int8][int8] | [int8][int8][int8][int8] | [int8][padding][padding][padding]|[int32]`

Reply:
Yes-yes, I acknowledged that right in my comment to Ahson above. I'll rearrange this.

Reply:
Done.

Reply:
Follow-up question: How are you figuring out how these struct fields are being laid out in memory to know that we are getting alignment? I understand the integer example, but what about array alignment, in C? Is that a pointer (word size) and int32 for the length?

Reply:
Array elements are always getting placed sequentially right next to each other so that `&array[0] + sizeof(array[0]) == &array[1]`. In this case specifically, we have an array of bytes, so it is going to be `_az_TOKEN_BUF_SIZE` number of bytes placed sequentially.

(`_az_TOKEN_BUF_SIZE` is 2048 bytes and I think it additionally might be a good practice to round byte array sizes to "binary round" numbers for this reason, although I am not completely sure enough to make that recommendation decisively - nothing bad would happen if it would be just 2000 bytes, but certainly it won't hurt if it is 2048. I personally like it for aesthetic reasons, and also with "binary round" numbers you can sometimes do tricks like bit masking, but again, not enough evidence to recommend it for all cases, other than personal preference)

Reply:
The compiler will properly align fields (guaranteed) [unless you do something special like this: https://docs.microsoft.com/en-us/cpp/preprocessor/pack?view=vs-2019] - there is nothing we have to do. A byte will be on a byte boundary, a 16-bit field will be on a 16-bit boundary, a 32-bit field will be on a 32-bit boundary, and a 64-bit value will be on a 64-bit boundary. A pointer is either 32- or 64-bits depending on whether you compile for 32-bit or 64-bits. An array will be aligned as per the array type. So, in this case, a byte array is aligned on a byte boundary. For an array, it doesn't matter if its 2000 versus 2048 elements -- we should pick the number we want for the array; not a number that looks appealing in source code.